docs/src/userguide/config/event_config.rst

```diff
@@ -126,13 +126,176 @@ Example configuration for a custom event input that is not supported by Klio:
 
     Any arbitrary key-value pairs for custom event input configuration specific to a job.
 
+.. _event-config-bigquery:
 
-.. todo:: document supported batch event inputs
+GoogleBigQuery
+^^^^^^^^^^^^^^
+
+*Mode: Batch*
+
+Consuming KlioMessages from `Google BigQuery`_ is supported for all runners. Klio allows the user to specify events from queries and from tables.
+
+Example configuration for `Google BigQuery`_, by specifying a table:
+
+    .. code-block:: yaml
+
+        name: my-cool-job
+        pipeline_options:
+          streaming: True
+        job_config:
+        events:
+            inputs:
+            - type: bq
+              columns:
+              - entity_id
+              dataset: bigquery_dataset
+              project: gcp_project
+              table: bigquery_table
 
 
+Example configuration for `Google BigQuery`_, by specifying a query in the BigQuery dialect:
+
+    .. code-block:: yaml
+
+        name: my-cool-job
+        pipeline_options:
+          streaming: True
+        job_config:
+        events:
+            inputs:
+            - type: bq
+              query:
+                SELECT * FROM [gcp_project.bigquery_dataset.bigquery_table]
+
+.. option:: job_config.events.inputs[].type
+
+    Value: ``bq``
+
+    | **Runner**: Dataflow, Direct
+    | *Required*
+
+.. option:: job_config.events.inputs[].columns[]
+
+    Not applicable if query is specified.
```

Comment:
Can you code format the word "query" to make it obvious you're referring to the config value? e.g.

```rst
``query``
```